static/app/views/insights/crons/components/checkInRow.tsx

```diff
@@ -0,0 +1,394 @@
+  ) : (
+    emptyCell
+  );
+
+  const groupsColumn =
+    groups && groups.length > 0 ? (
+      <IssuesContainer>
+        {groups.map(({id, shortId}) => (
+          <QuickContextHovercard
+            dataRow={{
+              ['issue.id']: id,
+              issue: shortId,
+            }}
+            contextType={ContextType.ISSUE}
+            organization={organization}
+            key={id}
+          >
+            <StyledShortId
+              shortId={shortId}
+              avatar={<ProjectBadge project={monitor.project} hideName avatarSize={12} />}
+              to={`/organizations/${organization.slug}/issues/${id}/`}
+            />
+          </QuickContextHovercard>
+        ))}
+      </IssuesContainer>
+    ) : (
+      emptyCell
+    );
+
+  return (
+    <Fragment>
+      {statusColumn}
+      {startedColumn}
+      {completedColumn}
+      {durationColumn}
+      {groupsColumn}
+      {hasMultiEnv ? <div>{environment}</div> : null}
+      {expectedTimeColum}
+    </Fragment>
+  );
+}
+
+interface OffScheduleIndicatorProps {
+  checkIn: CheckIn;
+}
+
+/**
+ * Renders a "Early" tag when the check-in occurred before it was expected.
+ *
+ * In scenarios where there is no in-progress and we know the duration of the
+ * job, we can suggest the problem is the missing in-progress
+ */
+function OffScheduleIndicator({checkIn}: OffScheduleIndicatorProps) {
+  const {duration, dateAdded, dateInProgress, expectedTime, monitorConfig} = checkIn;
+
+  const durationSeconds = duration && duration / 1000;
+  const marginSeconds = (monitorConfig.checkin_margin ?? DEFAULT_CHECKIN_MARGIN) * 60;
+
+  const beforeExpected = moment(dateAdded).isBefore(expectedTime);
+
+  // The check-in is on time if we're not checking-in before the expected
+  // check-in time. If we are after the expected check-in time this means the
+  // check-in happened before a miss was marked, and we are in the grace window.
+  if (!beforeExpected) {
+    return null;
+  }
+
+  const earlyBy = (
+    <strong>
+      <Duration seconds={moment(expectedTime).diff(dateAdded, 'seconds')} />
+    </strong>
+  );
+
+  const checkInMargin = (
+    <strong>
+      <Duration seconds={marginSeconds} />
+    </strong>
+  );
+
+  const jobDuration = durationSeconds && (
+    <strong>
+      <Duration seconds={durationSeconds} />
+    </strong>
+  );
+
+  // If the check-in ran longer than the configured margin and we're missing an
+  // in-progress check-in, it's very likely this check-in is off-schedule
+  // because of the missing in-progress.
+  const dueToMissingInProgress =
+    durationSeconds && durationSeconds > marginSeconds && !dateInProgress;
+
+  const title = dueToMissingInProgress
+    ? tct(
+        'This check-in was received [earlyBy] before it was expected. This may be due to the missing in-progress check-in, as your job reported a duration of [jobDuration] without sending an in-progress. Your grace period for the first check-in is [checkInMargin].',
```

Comment:
```suggestion
        'This check-in was recorded [earlyBy] earlier than expected. This may be due to a missing in-progress check-in, as your job reported a duration of [jobDuration] without sending one. The grace period for your monitor is [checkInMargin], before the check-in is considered missed.',
```

Reply:
We actually KNOW they have a missing in progress, which is why I had `the missing in-progress`. Wdyt?